test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -1647,6 +1654,396 @@ void test_iceCreateResponseForRequest_Controlled( void )
+
+/**
+ * @brief Validate ICE Handle Stun Packet functionality.
+ */
+void test_iceHandleStunPacket_FingerPrintMismatch( void )
+{
+    IceContext_t context = { 0 };
+    uint8_t stunMessageBuffer[] =
+    {
+        /* STUN header: Message Type = BINDING_SUCCESS_RESPONSE (0x0101), Length = 56 bytes (excluding 20 bytes header). */
+        0x01, 0x01, 0x00, 0x38,
+        /* Magic Cookie (0x2112A442). */
+        0x21, 0x12, 0xA4, 0x42,
+        /* 12 bytes (96 bits) transaction ID as generated by testRandomFxn. */
+        0x00, 0x01, 0x02, 0x03,0x04,  0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B,
+        /* Attribute type = XOR Mapped Address (0x0020), Length = 8 bytes. */
+        0x00, 0x20, 0x00, 0x08,
+        /* Address family = IPv4, Port = 0x3E82, IP Address = 0xC0A80164 (192.168.1.100). */
+        0x00, 0x01, 0x3E, 0x82,0xE1,  0xBA, 0xA5, 0x26,
+        /* Attribute type = ICE-CONTROLLED  ( 0x8029 ), Length = 8 bytes. */
+        0x80, 0x29, 0x00, 0x08,
+        /* Attribute Value = 0x0706050403020100. */
+        0x07, 0x06, 0x05, 0x04,0x03,  0x02, 0x01, 0x00,
+        /* Attribute type = MESSAGE-INTEGRITY (0x0008), Length = 20 bytes. */
+        0x00, 0x08, 0x00, 0x14,
+        /* Attribute Value = HMAC value as computed by testHmacFxn. */
+        0x6D, 0x6E, 0x63, 0x51,
+        0x4D, 0x42, 0xC5, 0x31,
+        0x73, 0x76, 0x6D, 0x71,
+        0x60, 0x69, 0x69, 0x64,
+        0x69, 0x65, 0x5A, 0x6A,
+        /* Attribute type = FINGERPRINT (0x8028), Length = 4 bytes. */
+        0x80, 0x28, 0x00, 0x04,
+        /* Attribute Actual Value: 0xBCD5BE24 as calculated by testCrc32Fxn whereas Attribute Vallue being Passed : 0xB1D5BE24 */
+        0xB1, 0xD5, 0xBE, 0x24       
+    };
+    size_t stunMessageBufferLength = sizeof( stunMessageBuffer );
+    IceEndpoint_t localEndpoint = { 0 };
+    IceEndpoint_t remoteEndpoint = { 0 };
+    uint8_t * pTransactionId;
+    IceCandidatePair_t * pCandidatePair;
+    IceResult_t resultIce;
+    IceHandleStunPacketResult_t result;
+
+    resultIce = Ice_Init( &( context ),
+                          &( initInfo ) );
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_OK,
+                       resultIce );
+
+    localEndpoint.isPointToPoint = 1;
+    localEndpoint.transportAddress.family = 0x01;
+    localEndpoint.transportAddress.port = 8080;
+    memcpy( ( void * ) &( localEndpoint.transportAddress.address[ 0 ] ),
+            ( const void * ) ipAddress,
+            sizeof( ipAddress ) );
+
+    remoteEndpoint = localEndpoint; /* For simplicity, use the same endpoint for remote */
+
+    result = Ice_HandleStunPacket( &( context ),
+                                   &( stunMessageBuffer[ 0 ] ), stunMessageBufferLength, &( localEndpoint ), &( remoteEndpoint ), &( pTransactionId ), &( pCandidatePair ) );
+
+    TEST_ASSERT_EQUAL( ICE_HANDLE_STUN_PACKET_RESULT_DESERIALIZE_ERROR,       /* Should have been ICE_HANDLE_STUN_PACKET_RESULT_FINGERPRINT_MISMATCH */
```

Comment:
What does the comment mean? Is the return value incorrect?

Reply:
I think it's helpful if ICE can return ICE_HANDLE_STUN_PACKET_RESULT_FINGERPRINT_MISMATCH instead of just ICE_HANDLE_STUN_PACKET_RESULT_DESERIALIZE_ERROR.